network/src/protocols/identity.rs

```diff
@@ -83,28 +84,24 @@ mod tests {
         let server_handshake_clone = server_handshake.clone();
         let client_handshake_clone = client_handshake.clone();
 
-        let server = async move {
-            let handshake = exchange_handshake(&server_handshake, &mut inbound)
-                .await
-                .expect("Handshake fails");
+        let server = handshake(&client_handshake_clone, &server_handshake, inbound);
+        let client = handshake(&server_handshake_clone, &client_handshake, outbound);
 
-            assert_eq!(
-                lcs::to_bytes(&handshake).unwrap(),
-                lcs::to_bytes(&client_handshake_clone).unwrap()
-            );
-        };
-
-        let client = async move {
-            let handshake = exchange_handshake(&client_handshake, &mut outbound)
-                .await
-                .expect("Handshake fails");
+        block_on(join(server, client));
+    }
 
-            assert_eq!(
-                lcs::to_bytes(&handshake).unwrap(),
-                lcs::to_bytes(&server_handshake_clone).unwrap()
-            );
-        };
+    async fn handshake(
+        remote_msg: &HandshakeMsg,
+        self_msg: &HandshakeMsg,
+        mut socket: MemorySocket,
+    ) {
```

Comment:
why the split here? if we're not reusing this function, the separation seems a little awkward as it doesn't reduce the the actual primary function's length much.

Reply:
CLion was just telling me it was duplicated, and I felt like collapsing it, but I can put it back.  It was actually helpful for me to understand `async` and how a bunch of borrowing around it worked.

Reply:
if we could merge, that would make me happy, but I'll leave that to you.

Reply:
Do you mean change it back by merge?  Or the changed version I have here?

Reply:
the original, i.e., no need for this extra handshake function afaiu.

Reply:
Changed back to the original.